cabby/data/metagraph/utils.py

```diff
@@ -103,11 +108,88 @@ def update_osm_map(osm_map: Map,
   print("In update_osm_map: %d nodes in graph after adding." % (
     osm_map.nx_graph.number_of_nodes()))
 
+
+def add_conceptual_nodes_and_edges(graph: nx.Graph,
+                                   poi_map: Dict[Text, int],
+                                   wd_relations: pd.DataFrame):
+  for _, row in wd_relations.iterrows():
+    place_node_id = poi_map[row["place"]]
+    graph.add_edge(row["instanceLabel"], place_node_id)
+    graph.add_edge(place_node_id, row["instanceLabel"])
+
+def construct_human_readable_name(poi_row: pd.Series) -> Text:
+    name = ""
+    if isinstance(poi_row["name"], str):
+        name = poi_row["name"]
+    elif isinstance(poi_row["wikipedia"], str):
+        name = poi_row["wikipedia"][3:]
+    elif isinstance(poi_row["brand"], str):
+        name = poi_row["brand"]
+    elif isinstance(poi_row["shop"], str):
+        name = poi_row["shop"]
+    elif isinstance(poi_row["tourism"], str):
+        name = poi_row["tourism"]
+    elif isinstance(poi_row["amenity"], str):
+        name = poi_row["amenity"]
+    return "%s_%s" % (name, poi_row["osmid"])
+
+def convert_multidi_to_weighted_undir_graph(
```

Comment:
Instead of iterating over edges, I think it would be preferable to iterate over nodes and their neighbors as shown in the networkx example here:

https://networkx.org/documentation/networkx-1.11/tutorial/tutorial.html#multigraphs

I think this would make it easier for us to choose other ways of aggregating weights from duplicated edges (e.g. min, max, sum, etc). WDYT?

Reply:
SG! Fixed by 5e679565cf8ef805e8484dbb5babfa27359e338e

Reply:
Nice! I like how you added the aggregator function to the function.